azure-pipelines/build-release.yml

```diff
@@ -83,6 +89,15 @@ stages:
         GIT_BRANCH: $(Build.SourceBranchName)
         GIT_COMMIT: $(Build.SourceVersion)
 
+    - script: |
+        python ci_scripts/generate_docs.py --output_dir $(System.DefaultWorkingDirectory)/docs_output
+      displayName: 'Generate documentation'
+      condition: and(succeeded(), eq(variables['generateDocs'], 'true'))
+
+    - publish: $(System.DefaultWorkingDirectory)/docs_output
+      artifact: Documentation
+      condition: and(succeeded(), eq(variables['generateDocs'], 'true'))
+
```

Comment:
Would it make sense to do this in a separate stage? There is an overhead of installing the python dependencies but is this cleaner? The point of the `uploadCoverage` was that we needed the unit test results and didn't want to upload the results multiple times.

Reply:
You are right I think it should be. I will set it up. 👍